javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll

```diff
@@ -118,57 +119,15 @@ module DataFlow {
 
     /** Gets a function value that may reach this node. */
     final FunctionNode getAFunctionValue() {
-      result = getAFunctionValue(_)
+      CallGraph::getAFunctionReference(result, 0).flowsTo(this)
```

Comment:
Why do we restrict imprecision to zero now? Wasn't it unrestricted before?

Reply:
This diff is relative to an earlier commit. On master, the imprecision bit wasn't easily available and I don't think it was a deliberate choice to include imprecise values here in the first place.

It seems to me that dealing with the imprecise callees is mainly for queries that rely on seeing all the potential callees, which has become a bit of a niche thing given that data flow queries usually don't do that.

But we should make sure it behaves the same way as `InvokeNode.getACallee()`, which defaults to _any_ imprecision level. I'd prefer to make the default imprecision level 0 for both 0-argument predicates, so you have to "opt in" to see the imprecise callees. WDYT?

Reply:
I'm not entirely sure. If we restrict ourselves to zero imprecision, what do we lose? Is it just calls to externs (modelling built-ins) and global functions defined elsewhere?

Reply:
Pretty much yes.

Reply:
Pushed a commit that makes `imprecision` zero the default for `getAFunctionValue` and for `getACallee`